hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/BulkInsertPartitioner.java

```diff
@@ -72,4 +79,37 @@ default Option<WriteHandleFactory> getWriteHandleFactory(int partitionId) {
     return Option.empty();
   }
 
+  /*
+   * If possible, we want to sort the data by partition path. Doing so will reduce the number of files written.
+   **/
+  static String[] prependPartitionPathColumn(String[] columnNames, HoodieWriteConfig config) {
+    if (config.getMetadataConfig().populateMetaFields()) {
```

Comment:
Let's unify both branches of this conditional: the only difference b/w them should be how we obtain the partition-field (meta-field vs configured one)